File: Data Structures & Algorithms/surrounded-regions/submission-0.cpp
class Solution {
public:
    void solve(vector<vector<char>>& board) {
        int ROWS = board.size(), COLS = board[0].size();
        this->ROWS = ROWS;
        this->COLS = COLS;
        
        for (int r = 0; r < ROWS; r++){
            for (int c = 0; c < COLS; c++){
                if ((r == 0 || c == 0 || r == ROWS - 1|| c == COLS - 1)
                    && board[r][c] == 'O'){
                    dfs(board, r, c);
                }
            }
        }

        for (int r = 0; r < ROWS; r++){
            for (int c = 0; c < COLS; c++){
                if (board[r][c] == 'O') board[r][c] = 'X';
            }
        }

        for (int r = 0; r < ROWS; r++){
            for (int c = 0; c < COLS; c++){
                if (board[r][c] == 'T') board[r][c] = 'O';
            }
        }
    }
private:
    int ROWS, COLS;
    void dfs(vector<vector<char>>& board, int r, int c) {
        if (r < 0 || r >= ROWS || 
            c < 0 || c >= COLS || 
            board[r][c] != 'O') return;

        board[r][c] = 'T';
        dfs(board, r + 1, c);
        dfs(board, r - 1, c);
        dfs(board, r, c + 1);
        dfs(board, r, c - 1);
    }
};

File: Data Structures & Algorithms/islands-and-treasure/submission-0.py
class Solution:
    def islandsAndTreasure(self, grid: List[List[int]]) -> None:
        visited = set()
        q = deque()
        ROWS, COLS = len(grid), len(grid[0])

        for r in range(ROWS):
            for c in range(COLS):
                if grid[r][c] == 0:
                    q.append([r, c])
                elif grid[r][c] == -1:
                    visited.add((r, c))

        level = 1
        dirs = [[1, 0], [-1, 0], [0, 1], [0, -1]]
        while q:
            for _ in range(len(q)):
                r, c = q.popleft()
                for dr, dc in dirs:
                    new_r, new_c = r + dr, c + dc
                    if (0 <= new_r < ROWS and 
                        0 <= new_c < COLS and
                        (new_r, new_c) not in visited and
                        grid[new_r][new_c] == 2147483647):

                        q.append([new_r, new_c])
                        visited.add((new_r, new_c))
                        grid[new_r][new_c] = level
                        
             
            level += 1
        return


        